File: FinalProjesi/app/src/main/java/com/example/flagquizapp/MainActivity.java
package com.example.flagquizapp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

import com.example.flagquizapp.R;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button flagQuizBtn = findViewById(R.id.flagQuizButton);
        Button capitalQuizBtn = findViewById(R.id.capitalQuizButton);
        Button generalKnowledgeBtn = findViewById(R.id.generalKnowledgeButton);
        Button mapQuizBtn = findViewById(R.id.mapQuizButton);

        flagQuizBtn.setOnClickListener(v -> {
            Intent intent = new Intent(this, QuizActivity.class);
            intent.putExtra("category", "flags_1bayrak4ulke"); // ← Veritabanındaki kategoriyle birebir aynı olmalı
            startActivity(intent);
        });

        capitalQuizBtn.setOnClickListener(v -> {
            Intent intent = new Intent(this, QuizActivity.class);
            intent.putExtra("category", "capitals");
            startActivity(intent);
        });

        generalKnowledgeBtn.setOnClickListener(v -> {
            Intent intent = new Intent(this, QuizActivity.class);
            intent.putExtra("category", "general_knowledge");
            startActivity(intent);
        });

        mapQuizBtn.setOnClickListener(v -> {
            Intent intent = new Intent(this, QuizActivity.class);
            intent.putExtra("category", "map_quiz");
            startActivity(intent);
        });
    }
}

File: FinalProjesi/app/src/main/java/com/example/flagquizapp/MainActivity.kt
package com.example.flagquizapp

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main) // enableEdgeToEdge()'i kaldırabiliriz

        // Butonları tanımlayalım
        val btnFlagQuiz = findViewById<Button>(R.id.btnFlagQuiz)
        val btnGeneralKnowledge = findViewById<Button>(R.id.btnGeneralKnowledge)
        val btnMapQuiz = findViewById<Button>(R.id.btnMapQuiz)
        val btnCapitalQuiz = findViewById<Button>(R.id.btnCapitalQuiz)

        // Ülke Bayrak Eşleştirme tıklandığında yeni ekrana geçiş
        btnFlagQuiz.setOnClickListener {
            startActivity(Intent(this, FlagQuizSelectionActivity::class.java))
        }

        // Genel Kültür quizi doğrudan başlatılacak
        btnGeneralKnowledge.setOnClickListener {
            startActivity(Intent(this, GeneralKnowledgeQuizActivity::class.java))
        }

        // Harita Quiz ekranına geçiş
        btnMapQuiz.setOnClickListener {
            startActivity(Intent(this, MapQuizSelectionActivity::class.java))
        }

        // Başkent Quiz ekranına geçiş
        btnCapitalQuiz.setOnClickListener {
            startActivity(Intent(this, CapitalQuizSelectionActivity::class.java))
        }
    }
}